### ISSUE TITLE
`docker top` should validate arguments for `ps`

### ISSUE THREAD

**Issue by AkihiroSuda** [docker/docker#24357]
**Output of `docker version`:**

    Client:
     Version:      1.12.0-rc2
     API version:  1.24
     Go version:   go1.6.2
     Git commit:   906eacd-unsupported
     Built:        Fri Jun 17 21:12:28 2016
     OS/Arch:      linux/amd64
     Experimental: true
    
    Server:
     Version:      1.12.0-rc2
     API version:  1.24
     Go version:   go1.6.2
     Git commit:   906eacd-unsupported
     Built:        Fri Jun 17 21:12:28 2016
     OS/Arch:      linux/amd64
     Experimental: true

**Output of `docker info`:**

    Containers: 2
     Running: 2
     Paused: 0
     Stopped: 0
    Images: 1
    Server Version: 1.12.0-rc2
    Storage Driver: overlay
     Backing Filesystem: extfs
    Logging Driver: json-file
    Cgroup Driver: cgroupfs
    Plugins:
     Volume: local
     Network: host bridge null overlay
    Swarm: inactive
    Runtimes: default
    Default Runtime: default
    Security Options: apparmor seccomp
    Kernel Version: 4.4.0-24-generic
    Operating System: Ubuntu 16.04 LTS
    OSType: linux
    Architecture: x86_64
    CPUs: 4
    Total Memory: 15.6 GiB
    Name: suda-PC2
    Docker Root Dir: /var/lib/docker
    Debug Mode (client): false
    Debug Mode (server): false
    Registry: https://index.docker.io/v1/
    WARNING: No swap limit support
    Experimental: true
    Insecure Registries:
     127.0.0.0/8

In Ubuntu 16.04, `/bin/ps` is provided by `procps_3.3.10-4ubuntu2_amd64.deb`.

**Steps to reproduce the issue:**

1. Suppose that a user `u1` has UID=1000.
2. Suppose that a container `c1` has a process of which PID is 1000 (in the host). Actually the PID is unlike to correspond with the UID of `u1`, but attacker can increase the probability by running `docker run` for just forking repeatedly. Attacker can also utilize GID or something else instead of UID.
3. Run `sudo -u u1 -E SECRET=foobar cat` _outside_ the container. (`cat` is not important. Any command that does not exit immediately should work.)
4. Run `docker top c1 ae -o uid=PID -o command=COMMAND`

**Describe the results you received:**

    PID                 COMMAND
    1000                cat SECRET=foobar

**Describe the results you expected:**

Host-side process should not be shown in `docker top`

**Additional information you deem important (e.g. issue happens only occasionally):**

If an user is allowed to run `docker top`, he/she is likely allowed to run `docker run --privilege` as well and do anything for entire the host.
So basically the issue is not critical in typical deployment.

(But it is not 100%: he/she may be only allowed to run some specific docker commands via `sudo` or some kind of RPC proxy.)

**Comment by AkihiroSuda** [comment]
opened PR #24358
